sycl/cmake/modules/AddSYCLUnitTest.cmake

```diff
@@ -73,6 +74,7 @@ macro(add_sycl_unittest_with_device test_dirname link_variant)
     -DGTEST_LANG_CXX11=1
     -DGTEST_HAS_TR1_TUPLE=0
     -D__SYCL_BUILD_SYCL_DLL
+    -DSYCL2020_DISABLE_DEPRECATION_WARNINGS
```

Comment:
Why is it required here?

Reply:
We still use SYCL 1.2.1 APIs inside runtime and tests. To avoid build failures due to `-Werror`, I suppressed these warnings for compiler build.

Reply:
Ok. Maybe I'm missing something, but is it a right level of Makefiles to add this flag?
It looks like it's been added for all LIT tests? Is it possible to add flag individually to tests which are really designed to test older 1.2.1 API rather than using it by accident?

I'd like us to stop adding more tests with 1.2.1 APIs.

Reply:
@pvchupin no, it's only for unit tests. We build them as part of compiler and link with the runtime statically, because many internal symbols are not exported from `libsycl.so` (to avoid accidental linkage of user code with our private interfaces). Other LIT tests (those in `sycl/test/` or llvm-test-suite) won't be affected by this option.

Reply:
I've updated tests to only disable warnings in the TUs, that use interop API